src/runtime/HalideBuffer.h

```diff
@@ -76,10 +76,15 @@ struct AllocationHeader {
 
 /** A similar struct for managing device allocations. */
 struct DeviceRefCount {
+    enum struct Ownership : int {
+        Allocated,
+        WrappedNative,
+        Unmanaged,
+    };
     // This is only ever constructed when there's something to manage,
     // so start at one.
     std::atomic<int> count {1};
-    bool wrapping_native_device_handle {false};
+    Ownership ownership;
```

Comment:
To match old semantics, this should have a default value of Allocated. (Or should we add a ctor to ensure that new instances are explicitly inited?)

Can this ever change for a given instance? If not, making it const and adding a ctor might be useful.